Prevent rename of visible levels

davidsbailey opened the pull request:
# Description

Prevent rename of levels in a visible script. Also prevent rename of levels which are loaded when you go to a url like /projects/applab or /projects/dance . 

## Links

- [jira](https://codedotorg.atlassian.net/browse/LP-294)

## Testing story

Manually verified the following:

1. no change in behavior for level in a hidden script
![Screen Shot 2019-11-25 at 6 44 20 PM](https://user-images.githubusercontent.com/8001765/69595308-0aec2700-0fb4-11ea-9a91-c21dbcebf657.png)

2. correct behavior + message for level in a visible script:
![Screen Shot 2019-11-25 at 6 44 27 PM](https://user-images.githubusercontent.com/8001765/69595334-17707f80-0fb4-11ea-8641-9ad9ebbc2266.png)

3. correct behavior + message for a standalone project level:
![Screen Shot 2019-11-25 at 6 44 35 PM](https://user-images.githubusercontent.com/8001765/69595351-21927e00-0fb4-11ea-9962-b4b521b24cc3.png)


# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [x] New features are translatable or updates will not break translations
- [x] Relevant documentation has been added or updated
- [x] User impact is well-understood and desirable
- [x] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

islemaster commented:
Is there a way to unit-test this? This seems like a bit of business logic we should cover. (Unless I've misunderstood the motivation for this change - it's not captured in the PR.)

dmcavoy commented on dashboard/app/views/levels/_form.html.haml:
```diff
@@ -1,102 +0,0 @@
-- content_for(:head) do
-  %script{src: webpack_asset_path('js/levelbuilder.js')}
-
-%pre#validation-error.validation-error{style: 'background-color: yellow; display: none'}
```
looks like you lost this line below. Not sure if intentional

davidsbailey replied:
Sorry, yes, this was intentional. I did this in a separate commit with a descriptive name, but forgot to mention it in the PR description.

davidsbailey commented:
> Is there a way to unit-test this? This seems like a bit of business logic we should cover. (Unless I've misunderstood the motivation for this change - it's not captured in the PR.)

Good call. I've added unit tests. The motivation, which is not really explained in the linked jira ticket, is to prevent levelbuilders from editing levels in cases where that could cause progress loss, which includes any time a script is visible. I added on the bit about standalone project levels because it seems like changing the name of those would break the corresponding standalone project page.

In explaining this I realized that I've failed to cover the case for pilot scripts, which are always marked hidden. I'll add that now. Thank you for prompting me for this!
